### ISSUE TITLE
character relationship component

### ISSUE THREAD

**Issue by joshdmiller** [StoryShop/app#24]
## Overview

Create a component to display a list of characters and their associated relationship descriptions, with actions to edit and remove.

Modified from the [existing source](https://github.com/StoryShop/app/blob/3736047a8f399cc840809481f1cd97abe6e8e7f3/src/components/characters/relationships/index.js).
## Component Signature

``` js
type Props = {
  relationships: Array<Relation>,

  onDelete: ?( _id: string ) => void,
  onEdit: ?( _id: string, description: string ) => void,
};

type Relation = {
  _id: string,
  avatar: string.
  name: string,
  description: string,
};
```
## UI

Renders a MaterialUI `List`, with a single `ListItem` for each relationship in the list. Each `ListItem` has a left avatar, defaulting to the `User` material icon, primary text of the character's name, secondary text of the relationship description, and a secondary/left action of a delete icon.
### Primary Action

The primary action is a modal dialog prompt to edit the description text. On success, the `onEdit` callback should be called.
### Secondary Action

The secondary action is a delete icon. On click, the `onDelete` callback should be called.

**Comment by chrisharrington** [comment]
Where is this component supposed to go in the app? Is it supposed to go on the existing character page to show relationships to other characters or..?

**Comment by joshdmiller** [comment]
@chrisharrington All react components will sit in isolation. At some point, we'll add it to the character route, but that also requires changes to the model mapping, dispatch mapping, and the falcor routes. So for now, just a component, some unit tests, and a react story. 😄

**Comment by chrisharrington** [comment]
@josh Interesting. Is there a test page or something into which I can render the component to see how it looks?

**Comment by joshdmiller** [comment]
Yeah, that's the story. Check out my comment in the #general slack channel from earlier today for details and an example, and check out the React Storybook for more docs.

**Comment by chrisharrington** [comment]
@joshdmiller What's an example of the relationship description? One or two lines?

**Comment by chrisharrington** [comment]
@joshdmiller Can the relationship description be empty? Are there any other validation rules I should worry about? Maximum length?

**Comment by joshdmiller** [comment]
@chrisharrington Examples: Father, Brother, Daughter, Mother, Friend, Enemy, guy she met in a bar. I'm not too worried about any length; I'd rather see how users use the field before we put an arbitrary restriction on it. I think relationship description should probably be required, but that might be an assumption on my part. On the backend, we will enforce uniqueness on the name between two characters (e.g. Bob and Fred can have multiple relationships with each other, but not multiple "Father" relationships).

cc @sethmerrick

**Comment by joshdmiller** [comment]
Let's make sure PRs in the future use the proper `Closes` footer in the commit so the issues close when the PRs are merged.

Closed by #112.
